src/Microsoft.Identity.Client/AppConfig/AbstractApplicationBuilder.cs

```diff
@@ -37,6 +38,41 @@ public T WithHttpClientFactory(IMsalHttpClientFactory httpClientFactory)
             return (T)this;
         }
 
+        /// <summary>
+        /// Allows developers to configure their own valid authorities. A json string similar to https://aka.ms/aad-instance-discovery should be provided.
+        /// MSAL uses this information to: 
+        /// <list type="bullet">
+        /// <item>Call REST APIs on the environment specified in the preferred_network</item>
+        /// <item>Identify an environment under which to save tokens and accounts in the cache</item>
+        /// <item>Use the environment aliases to match tokens issued to other authorities</item>
+        /// </list>
+        /// For more details see https://aka.ms/msal-net-custom-instance-metadata
+        /// </summary>
+        /// <remarks>
+        /// Developers take responsibility for authority validation if they use this method. Should not be used when the authority is not know in advance. 
+        /// Has no effect on ADFS or B2C authorities, only for AAD authorities</remarks>
+        /// <param name="instanceDiscoveryJson"></param>
+        /// <returns></returns>
+        public T WithInstanceDicoveryMetadata(string instanceDiscoveryJson)
+        {
+            if (string.IsNullOrEmpty(instanceDiscoveryJson))
```

Comment:
please use { } even on one-liners

Reply:
Actually our new code style, which is copied from the .net core guys, specifically enables this. It is the reason we moved to it - @brentschmaltz uses this in their projects and we wanted to have the same code standard across C# libs.

Reply:
nope.  totally disagree.  this causes bugs and reduces readability.  we need to revert that change.

Reply:
I agree with @MarkZuber on that (and disagree with Brent). New comers coming to the code or PMs won't pay attention. Having the {} on one lines makes it safer to change the block

Reply:
Ok, I don't have strong opinions on this particular item, but we had a conversation over email on this and made this decisions. Please reopen that conversation or start a new one,

Reply:
Personally I disagree that is reduces readability - because you can fit more important lines of code on the screen, i.e. 

```csharp 
if (arg == null) 
{
   throw new ArgumentNullException("arg");
}
```
is less readable than

```csharp
if (arg == null) 
   throw new ArgumentNullException("arg");
```

I agree it is potentially more dangerous for developers who are not familiar with C# / C / Java.